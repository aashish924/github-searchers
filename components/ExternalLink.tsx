import React, {CSSProperties} from 'react'
import styled from 'styled-components';


const A = styled.a`
  color: #008c99;
  text-decoration: none;
  padding: .2em .8em;

  &:hover {
    color: #0ab0bf;
  }
  &:active {
    color: #70ccd3;
  }
  &:visited {
    color: #666666;
  }
`;


const ExternalLink: React.FC<{url: string, label: string, style?: CSSProperties}> = ({url, label, style}) => {
  return <A href={url} target="_blank" style={style}>{label}</A>
};


export default ExternalLink;