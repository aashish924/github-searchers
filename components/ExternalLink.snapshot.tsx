import * as React from 'react';
import renderer from 'react-test-renderer';

import ExternalLink from './ExternalLink';

describe('ExternalLink should render correctly', () => {
    it('ExternalLink without custom styles', () => {
        const tree = renderer
            .create(<ExternalLink url="https://github-searchers.vercel.app" label="Github Searchers" />)
            .toJSON();
        expect(tree).toMatchSnapshot('without-custom-style');
    });

    it('ExternalLink without custom styles', () => {
        const tree = renderer
            .create(
                <ExternalLink
                    url="https://github-searchers.vercel.app"
                    label="Github Searchers"
                    style={{ color: 'red' }}
                />,
            )
            .toJSON();
        expect(tree).toMatchSnapshot('with-custom-style');
    });
});
