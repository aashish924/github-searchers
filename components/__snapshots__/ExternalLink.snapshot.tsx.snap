// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink should render correctly ExternalLink without custom styles: with-custom-style 1`] = `
<a
  className="sc-AxjAm kebPqi"
  href="https://github-searchers.vercel.app"
  style={
    Object {
      "color": "red",
    }
  }
  target="_blank"
>
  Github Searchers
</a>
`;

exports[`ExternalLink should render correctly ExternalLink without custom styles: without-custom-style 1`] = `
<a
  className="sc-AxjAm kebPqi"
  href="https://github-searchers.vercel.app"
  target="_blank"
>
  Github Searchers
</a>
`;
